listenbrainz/db/dump.py

```diff
@@ -226,38 +269,43 @@ def _create_dump(location, dump_type, tables, dump_time, threads=DUMP_DEFAULT_TH
                 tar.add(DUMP_LICENSE_FILE_PATH,
                         arcname=os.path.join(archive_name, "COPYING"))
             except IOError as e:
-                current_app.logger.error('IOError while adding dump metadata: %s', str(e), exc_info=True)
+                current_app.logger.error(
+                    'IOError while adding dump metadata: %s', str(e), exc_info=True)
                 raise
             except Exception as e:
-                current_app.logger.error('Exception while adding dump metadata: %s', str(e), exc_info=True)
+                current_app.logger.error(
+                    'Exception while adding dump metadata: %s', str(e), exc_info=True)
                 raise
 
-
             archive_tables_dir = os.path.join(temp_dir, 'lbdump', 'lbdump')
             create_path(archive_tables_dir)
 
-
             with db.engine.connect() as connection:
-                with connection.begin() as transaction:
-                    cursor = connection.connection.cursor()
-                    for table in tables:
-                        try:
-                            copy_table(
-                                cursor=cursor,
-                                location=archive_tables_dir,
-                                columns=','.join(tables[table]),
-                                table_name=table,
-                            )
-                        except IOError as e:
-                            current_app.logger.error('IOError while copying table %s', table, exc_info=True)
-                            raise
-                        except Exception as e:
-                            current_app.logger.error('Error while copying table %s: %s', table, str(e), exc_info=True)
-                            raise
-                    transaction.rollback()
-
-
-            tar.add(archive_tables_dir, arcname=os.path.join(archive_name, 'lbdump'.format(dump_type)))
+                if dump_type == "feedback":
+                    dump_user_feedback(connection, location=archive_tables_dir)
+                else:
+                    with connection.begin() as transaction:
```

Comment:
if we get one more dump type here I'd like to abstract this section out into a separate file too like you've done with `dump_user_feedback`, to aid in readability. For now it's probably OK though